ast-checks/src/comment_finders.rs

```diff
@@ -0,0 +1,57 @@
+use rnix::{NodeOrToken, SyntaxKind::*, SyntaxNode};
+
+use crate::tree_utils::{next_siblings, prev_siblings, walk_kind};
+
+
+pub fn find_comment_within(node: &SyntaxNode) -> Option<String> {
+    let mut maybe_comment_token = None;
+    for item in walk_kind(node, TOKEN_COMMENT) {
+        match item {
+            NodeOrToken::Token(token) => {
+                maybe_comment_token = Some(token);
+                break;
+            }
+            NodeOrToken::Node(_) => {
+                unreachable!("");
+            }
+        }
+    }
```

Comment:
You should rewrite this to use iterator methods: https://doc.rust-lang.org/std/iter/trait.Iterator.html